programs/mango-v4/src/instructions/serum3_place_order.rs

```diff
@@ -258,6 +258,48 @@ pub fn serum3_place_order(
         None
     };
 
+    max_base_qty = if serum_market.is_reduce_only() {
+        require!(
+            load_open_orders_ref(&ctx.accounts.open_orders.as_ref())?.free_slot_bits == u128::MAX,
+            MangoError::SomeError
+        );
+
+        let bank = ctx.accounts.payer_bank.load()?;
+        let token_pos_native = account
+            .token_position(bank.token_index)?
+            .native(&bank)
+            .to_num::<i64>();
+        let base_lot_size = load_market_state(
+            &ctx.accounts.serum_market_external,
+            &ctx.accounts.serum_program.key(),
+        )?
+        .coin_lot_size;
+        let max_base_qty_native = max_base_qty * base_lot_size;
+        require!(
+            max_base_qty_native >= token_pos_native.unsigned_abs(),
```

Comment:
I don't get this condition. Sounds like if the user has 10 base native and they want to buy 9 more that's an error, but buying 11 is fine? Can you explain?

Reply:
sorry it should be, 

```
max_base_qty_native <= token_pos_native.unsigned_abs()
```

I have fixed it now, intention is, that you cant place an order greater than your position, when market is in reduce only mode, see https://github.com/blockworks-foundation/mango-v3/blob/main/program/src/processor.rs#L2177